Sources/PropertyEditor.swift

```diff
@@ -0,0 +1,680 @@
+import Result
+
+/// A mutable, observable property that has an optionally failable action
+/// associated with the setter.
+///
+/// ## Consistency when nested
+///
+/// `PropertyEditor` would back-propagate the validation failure if a
+/// proposed value from an outer property fails with its inner property.
+///
+/// It would also evaluate values originated from the inner property. In other
+/// words, it is possible for `value` to be invalid, causing `validations`
+/// to be a `failure`. Rely on `validations` for asserting a pass in validation.
+public final class PropertyEditor<Value, ValidationError: Error> {
+	/// The commited value of the property.
+	///
+	/// It does not guarantee that the value is valid with regard to `self`, since
+	/// changes might be initiated from the inner properties. Check `validations`
+	/// for the latest validation state.
+	public var commited: Property<Value>
+
+	/// The lifetime of the property.
+	public let lifetime: Lifetime
+
+	/// The latest validation that have been made by the property.
+	public let result: Property<ValidationResult<Value, ValidationError>>
+
+	/// The action associated with the property.
+	private let action: (Value) -> Void
+
+	/// The existential box that wraps the synchronization mechanic of the root
+	/// property of the composed chain.
+	private let rootBox: PropertyEditorBoxBase<()>
```

Comment:
Both the variable name and comment leave me confused after reading. I can only tell from usage that rootBox appears to guard accesses/changes to the originating property.